<a href="https://colab.research.google.com/github/beyzaturku/2209/blob/main/alt_split.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import zipfile
import os

zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    file_list = zip_ref.namelist()
    train_ann_files = [file for file in file_list if 'train/ann' in file and file.endswith('.json')]

print(f"Toplam {len(train_ann_files)} JSON dosyası bulundu.")

Toplam 24143 JSON dosyası bulundu.


## Hight Alt

### train/ann klasöründe high etiketine sahip json dosyaları

In [ ]:
import json

# "high alt" etiketi içeren dosyaları bul
high_alt_files = []

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for ann_file in train_ann_files:
        # JSON dosyasını oku
        with zip_ref.open(ann_file) as file:
            data = json.load(file)
            # "high alt" etiketini kontrol et
            if any(tag['name'] == 'high alt' for tag in data.get('tags', [])):
                high_alt_files.append(ann_file)

# "high alt" etiketine sahip dosyaları listele
print("High alt etiketine sahip JSON dosyaları:", high_alt_files)
print(f"Toplam {len(high_alt_files)} JSON dosyası bulundu.")

High alt etiketine sahip JSON dosyaları: ['UAVDT/train/ann/M0501_img000001.jpg.json', 'UAVDT/train/ann/M0501_img000002.jpg.json', 'UAVDT/train/ann/M0501_img000003.jpg.json', 'UAVDT/train/ann/M0501_img000004.jpg.json', 'UAVDT/train/ann/M0501_img000005.jpg.json', 'UAVDT/train/ann/M0501_img000006.jpg.json', 'UAVDT/train/ann/M0501_img000007.jpg.json', 'UAVDT/train/ann/M0501_img000008.jpg.json', 'UAVDT/train/ann/M0501_img000009.jpg.json', 'UAVDT/train/ann/M0501_img000010.jpg.json', 'UAVDT/train/ann/M0501_img000011.jpg.json', 'UAVDT/train/ann/M0501_img000012.jpg.json', 'UAVDT/train/ann/M0501_img000013.jpg.json', 'UAVDT/train/ann/M0501_img000014.jpg.json', 'UAVDT/train/ann/M0501_img000015.jpg.json', 'UAVDT/train/ann/M0501_img000016.jpg.json', 'UAVDT/train/ann/M0501_img000017.jpg.json', 'UAVDT/train/ann/M0501_img000018.jpg.json', 'UAVDT/train/ann/M0501_img000019.jpg.json', 'UAVDT/train/ann/M0501_img000020.jpg.json', 'UAVDT/train/ann/M0501_img000021.jpg.json', 'UAVDT/train/ann/M0501_img000022.j

### dosyalardan .json uzantısı kaldırıldı.

In [ ]:
for ann_file in high_alt_files:
  image_name = ann_file.replace('UAVDT/train/ann/', '').replace('.jpg.json', '') + '.jpg'

In [ ]:
new_high_images= []

for ann_file in high_alt_files:
  image_name = ann_file.replace('UAVDT/train/ann/', '').replace('.jpg.json', '') + '.jpg'
  new_high_images.append(image_name)

print(f"Toplam {len(new_high_images)} dosya bulundu.")

Toplam 352 dosya bulundu.


In [ ]:
# "train/img" klasöründeki tüm dosyaları listele
img_files = []

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    img_files = [file for file in zip_ref.namelist() if 'train/img' in file and file.endswith('.jpg')]
# 'UAVDT/train/img/' kısmını her dosya isminden kaldır
img_file_names = [file.replace('UAVDT/train/img/', '') for file in img_files]

# Sonuçları yazdır
print("Dosya isimleri:", img_file_names)
print(f"Toplam {len(img_file_names)} görsel ismi bulundu.")

Dosya isimleri: ['M0101_img000001.jpg', 'M0101_img000002.jpg', 'M0101_img000003.jpg', 'M0101_img000004.jpg', 'M0101_img000005.jpg', 'M0101_img000006.jpg', 'M0101_img000007.jpg', 'M0101_img000008.jpg', 'M0101_img000009.jpg', 'M0101_img000010.jpg', 'M0101_img000011.jpg', 'M0101_img000012.jpg', 'M0101_img000013.jpg', 'M0101_img000014.jpg', 'M0101_img000015.jpg', 'M0101_img000016.jpg', 'M0101_img000017.jpg', 'M0101_img000018.jpg', 'M0101_img000019.jpg', 'M0101_img000020.jpg', 'M0101_img000021.jpg', 'M0101_img000022.jpg', 'M0101_img000023.jpg', 'M0101_img000024.jpg', 'M0101_img000025.jpg', 'M0101_img000026.jpg', 'M0101_img000027.jpg', 'M0101_img000028.jpg', 'M0101_img000029.jpg', 'M0101_img000030.jpg', 'M0101_img000031.jpg', 'M0101_img000032.jpg', 'M0101_img000033.jpg', 'M0101_img000034.jpg', 'M0101_img000035.jpg', 'M0101_img000036.jpg', 'M0101_img000037.jpg', 'M0101_img000038.jpg', 'M0101_img000039.jpg', 'M0101_img000040.jpg', 'M0101_img000041.jpg', 'M0101_img000042.jpg', 'M0101_img000043.

In [ ]:
matching_images = []

for img_name in new_high_images:
  if img_name in img_file_names:
    matching_images.append(img_name)

print(f"Toplam {len(matching_images)} dosya bulundu.")
print("Eşleşen dosyalar:", matching_images)

Toplam 352 dosya bulundu.
Eşleşen dosyalar: ['M0501_img000001.jpg', 'M0501_img000002.jpg', 'M0501_img000003.jpg', 'M0501_img000004.jpg', 'M0501_img000005.jpg', 'M0501_img000006.jpg', 'M0501_img000007.jpg', 'M0501_img000008.jpg', 'M0501_img000009.jpg', 'M0501_img000010.jpg', 'M0501_img000011.jpg', 'M0501_img000012.jpg', 'M0501_img000013.jpg', 'M0501_img000014.jpg', 'M0501_img000015.jpg', 'M0501_img000016.jpg', 'M0501_img000017.jpg', 'M0501_img000018.jpg', 'M0501_img000019.jpg', 'M0501_img000020.jpg', 'M0501_img000021.jpg', 'M0501_img000022.jpg', 'M0501_img000023.jpg', 'M0501_img000024.jpg', 'M0501_img000025.jpg', 'M0501_img000026.jpg', 'M0501_img000027.jpg', 'M0501_img000028.jpg', 'M0501_img000029.jpg', 'M0501_img000030.jpg', 'M0501_img000031.jpg', 'M0501_img000032.jpg', 'M0501_img000033.jpg', 'M0501_img000034.jpg', 'M0501_img000035.jpg', 'M0501_img000036.jpg', 'M0501_img000037.jpg', 'M0501_img000038.jpg', 'M0501_img000039.jpg', 'M0501_img000040.jpg', 'M0501_img000041.jpg', 'M0501_img00

### High alt etiketine sahip 352 görsel için ayrı klasör oluştur

In [ ]:
import os
import zipfile
from io import BytesIO
from PIL import Image

# Yeni klasör adı
new_folder = '/content/drive/MyDrive/2209/high_alt'

# Eğer klasör yoksa oluştur
if not os.path.exists(new_folder):
    os.makedirs(new_folder)

# Zip dosyasının yolu
zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

# Zip dosyasını aç ve görselleri al
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for img_name in matching_images:
        # Görsel dosyasının yolu
        img_path = f"UAVDT/train/img/{img_name}"

        # Eğer dosya zip dosyasında varsa
        if img_path in zip_ref.namelist():
            # Dosyayı okuma
            with zip_ref.open(img_path) as file:
                img_data = file.read()

            # Görseli bellekte aç
            img = Image.open(BytesIO(img_data))

            # Yeni dosya yolunu oluştur
            new_img_path = os.path.join(new_folder, img_name)

            # Görseli kaydet
            img.save(new_img_path)

            print(f"{img_name} kaydedildi.")

print(f"Tüm eşleşen görseller '{new_folder}' klasörüne kaydedildi.")
print(f"Toplam {len(matching_images)} görsel kaydedildi.")

M0501_img000001.jpg kaydedildi.
M0501_img000002.jpg kaydedildi.
M0501_img000003.jpg kaydedildi.
M0501_img000004.jpg kaydedildi.
M0501_img000005.jpg kaydedildi.
M0501_img000006.jpg kaydedildi.
M0501_img000007.jpg kaydedildi.
M0501_img000008.jpg kaydedildi.
M0501_img000009.jpg kaydedildi.
M0501_img000010.jpg kaydedildi.
M0501_img000011.jpg kaydedildi.
M0501_img000012.jpg kaydedildi.
M0501_img000013.jpg kaydedildi.
M0501_img000014.jpg kaydedildi.
M0501_img000015.jpg kaydedildi.
M0501_img000016.jpg kaydedildi.
M0501_img000017.jpg kaydedildi.
M0501_img000018.jpg kaydedildi.
M0501_img000019.jpg kaydedildi.
M0501_img000020.jpg kaydedildi.
M0501_img000021.jpg kaydedildi.
M0501_img000022.jpg kaydedildi.
M0501_img000023.jpg kaydedildi.
M0501_img000024.jpg kaydedildi.
M0501_img000025.jpg kaydedildi.
M0501_img000026.jpg kaydedildi.
M0501_img000027.jpg kaydedildi.
M0501_img000028.jpg kaydedildi.
M0501_img000029.jpg kaydedildi.
M0501_img000030.jpg kaydedildi.
M0501_img000031.jpg kaydedildi.
M0501_im

### json dosyaları için de ayrı klasör oluştur.

In [ ]:
import os
import zipfile

# Yeni klasör adı
new_ann_folder = '/content/drive/MyDrive/2209/high_alt_annotations'

# Eğer klasör yoksa oluştur
if not os.path.exists(new_ann_folder):
    os.makedirs(new_ann_folder)

# Zip dosyasının yolu
zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

# Zip dosyasını aç ve JSON dosyalarını al
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for img_name in matching_images:
        # JSON dosyasının yolu
        ann_file_name = img_name.replace('.jpg', '.jpg.json')
        ann_file_path = f"UAVDT/train/ann/{ann_file_name}"

        # Eğer dosya zip dosyasında varsa
        if ann_file_path in zip_ref.namelist():
            # Dosyayı okuma
            with zip_ref.open(ann_file_path) as file:
                ann_data = file.read()

            # Yeni dosya yolunu oluştur
            new_ann_path = os.path.join(new_ann_folder, ann_file_name)

            # JSON dosyasını kaydet
            with open(new_ann_path, 'wb') as new_file:
                new_file.write(ann_data)

            print(f"{ann_file_name} kaydedildi.")

print(f"Tüm eşleşen JSON dosyaları '{new_ann_folder}' klasörüne kaydedildi.")

M0501_img000001.jpg.json kaydedildi.
M0501_img000002.jpg.json kaydedildi.
M0501_img000003.jpg.json kaydedildi.
M0501_img000004.jpg.json kaydedildi.
M0501_img000005.jpg.json kaydedildi.
M0501_img000006.jpg.json kaydedildi.
M0501_img000007.jpg.json kaydedildi.
M0501_img000008.jpg.json kaydedildi.
M0501_img000009.jpg.json kaydedildi.
M0501_img000010.jpg.json kaydedildi.
M0501_img000011.jpg.json kaydedildi.
M0501_img000012.jpg.json kaydedildi.
M0501_img000013.jpg.json kaydedildi.
M0501_img000014.jpg.json kaydedildi.
M0501_img000015.jpg.json kaydedildi.
M0501_img000016.jpg.json kaydedildi.
M0501_img000017.jpg.json kaydedildi.
M0501_img000018.jpg.json kaydedildi.
M0501_img000019.jpg.json kaydedildi.
M0501_img000020.jpg.json kaydedildi.
M0501_img000021.jpg.json kaydedildi.
M0501_img000022.jpg.json kaydedildi.
M0501_img000023.jpg.json kaydedildi.
M0501_img000024.jpg.json kaydedildi.
M0501_img000025.jpg.json kaydedildi.
M0501_img000026.jpg.json kaydedildi.
M0501_img000027.jpg.json kaydedildi.
M

In [ ]:
import os

# Klasörlerin yolu
annotations_folder = '/content/drive/MyDrive/2209/high_alt_annotations'
images_folder = '/content/drive/MyDrive/2209/high_alt'

# Klasörlerdeki dosya isimlerini al
annotations_files = [f for f in os.listdir(annotations_folder) if os.path.isfile(os.path.join(annotations_folder, f))]
images_files = [f for f in os.listdir(images_folder) if os.path.isfile(os.path.join(images_folder, f))]

# 1. Klasörlerdeki dosya sayısını yazdır
print(f"high_alt_annotations klasöründe {len(annotations_files)} dosya bulundu.")
print(f"high_alt_images klasöründe {len(images_files)} dosya bulundu.")

high_alt_annotations klasöründe 352 dosya bulundu.
high_alt_images klasöründe 352 dosya bulundu.


## Medium Alt

In [ ]:
import json

# "medium alt" etiketi içeren dosyaları bul
medium_alt_files = []

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for ann_file in train_ann_files:
        # JSON dosyasını oku
        with zip_ref.open(ann_file) as file:
            data = json.load(file)
            # "medium alt" etiketini kontrol et
            if any(tag['name'] == 'medium alt' for tag in data.get('tags', [])):
                medium_alt_files.append(ann_file)

# "medium alt" etiketine sahip dosyaları listele
print("Medium alt etiketine sahip JSON dosyaları:", medium_alt_files)
print(f"Toplam {len(medium_alt_files)} JSON dosyası bulundu.")

Medium alt etiketine sahip JSON dosyaları: ['UAVDT/train/ann/M0101_img000001.jpg.json', 'UAVDT/train/ann/M0101_img000002.jpg.json', 'UAVDT/train/ann/M0101_img000003.jpg.json', 'UAVDT/train/ann/M0101_img000004.jpg.json', 'UAVDT/train/ann/M0101_img000005.jpg.json', 'UAVDT/train/ann/M0101_img000006.jpg.json', 'UAVDT/train/ann/M0101_img000007.jpg.json', 'UAVDT/train/ann/M0101_img000008.jpg.json', 'UAVDT/train/ann/M0101_img000009.jpg.json', 'UAVDT/train/ann/M0101_img000010.jpg.json', 'UAVDT/train/ann/M0101_img000011.jpg.json', 'UAVDT/train/ann/M0101_img000012.jpg.json', 'UAVDT/train/ann/M0101_img000013.jpg.json', 'UAVDT/train/ann/M0101_img000014.jpg.json', 'UAVDT/train/ann/M0101_img000015.jpg.json', 'UAVDT/train/ann/M0101_img000016.jpg.json', 'UAVDT/train/ann/M0101_img000017.jpg.json', 'UAVDT/train/ann/M0101_img000018.jpg.json', 'UAVDT/train/ann/M0101_img000019.jpg.json', 'UAVDT/train/ann/M0101_img000020.jpg.json', 'UAVDT/train/ann/M0101_img000021.jpg.json', 'UAVDT/train/ann/M0101_img000022

In [ ]:
new_medium_images= []

for ann_file in medium_alt_files:
  image_name = ann_file.replace('UAVDT/train/ann/', '').replace('.jpg.json', '') + '.jpg'
  new_medium_images.append(image_name)

print(f"Toplam {len(new_medium_images)} dosya bulundu.")

Toplam 16134 dosya bulundu.


In [ ]:
# "train/img" klasöründeki tüm dosyaları listele
img_files = []

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    img_files = [file for file in zip_ref.namelist() if 'train/img' in file and file.endswith('.jpg')]

# 'UAVDT/train/img/' kısmını her dosya isminden kaldır
img_file_names = [file.replace('UAVDT/train/img/', '') for file in img_files]

# Sonuçları yazdır
print("Dosya isimleri:", img_file_names)
print(f"Toplam {len(img_file_names)} görsel ismi bulundu.")

matching_images = []

for img_name in new_medium_images:
  if img_name in img_file_names:
    matching_images.append(img_name)

print(f"Toplam {len(matching_images)} dosya bulundu.")
print("Eşleşen dosyalar:", matching_images)

Dosya isimleri: ['M0101_img000001.jpg', 'M0101_img000002.jpg', 'M0101_img000003.jpg', 'M0101_img000004.jpg', 'M0101_img000005.jpg', 'M0101_img000006.jpg', 'M0101_img000007.jpg', 'M0101_img000008.jpg', 'M0101_img000009.jpg', 'M0101_img000010.jpg', 'M0101_img000011.jpg', 'M0101_img000012.jpg', 'M0101_img000013.jpg', 'M0101_img000014.jpg', 'M0101_img000015.jpg', 'M0101_img000016.jpg', 'M0101_img000017.jpg', 'M0101_img000018.jpg', 'M0101_img000019.jpg', 'M0101_img000020.jpg', 'M0101_img000021.jpg', 'M0101_img000022.jpg', 'M0101_img000023.jpg', 'M0101_img000024.jpg', 'M0101_img000025.jpg', 'M0101_img000026.jpg', 'M0101_img000027.jpg', 'M0101_img000028.jpg', 'M0101_img000029.jpg', 'M0101_img000030.jpg', 'M0101_img000031.jpg', 'M0101_img000032.jpg', 'M0101_img000033.jpg', 'M0101_img000034.jpg', 'M0101_img000035.jpg', 'M0101_img000036.jpg', 'M0101_img000037.jpg', 'M0101_img000038.jpg', 'M0101_img000039.jpg', 'M0101_img000040.jpg', 'M0101_img000041.jpg', 'M0101_img000042.jpg', 'M0101_img000043.

In [ ]:
import os
import zipfile
from io import BytesIO
from PIL import Image

# Yeni klasör adı
new_folder = '/content/drive/MyDrive/2209/medium_alt/medium_alt_images'

# Eğer klasör yoksa oluştur
if not os.path.exists(new_folder):
    os.makedirs(new_folder)

# Zip dosyasının yolu
zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

# Zip dosyasını aç ve görselleri al
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for img_name in matching_images:
        # Görsel dosyasının yolu
        img_path = f"UAVDT/train/img/{img_name}"

        # Eğer dosya zip dosyasında varsa
        if img_path in zip_ref.namelist():
            # Dosyayı okuma
            with zip_ref.open(img_path) as file:
                img_data = file.read()

            # Görseli bellekte aç
            img = Image.open(BytesIO(img_data))

            # Yeni dosya yolunu oluştur
            new_img_path = os.path.join(new_folder, img_name)

            # Görseli kaydet
            img.save(new_img_path)

            print(f"{img_name} kaydedildi.")

print(f"Tüm eşleşen görseller '{new_folder}' klasörüne kaydedildi.")
print(f"Toplam {len(matching_images)} görsel kaydedildi.")

Görüntülenen çıkış son 5000 satıra kısaltıldı.
M1002_img000342.jpg kaydedildi.
M1002_img000343.jpg kaydedildi.
M1002_img000344.jpg kaydedildi.
M1002_img000345.jpg kaydedildi.
M1002_img000346.jpg kaydedildi.
M1002_img000347.jpg kaydedildi.
M1002_img000348.jpg kaydedildi.
M1002_img000349.jpg kaydedildi.
M1002_img000350.jpg kaydedildi.
M1002_img000351.jpg kaydedildi.
M1002_img000352.jpg kaydedildi.
M1002_img000353.jpg kaydedildi.
M1002_img000354.jpg kaydedildi.
M1002_img000355.jpg kaydedildi.
M1002_img000356.jpg kaydedildi.
M1002_img000357.jpg kaydedildi.
M1002_img000358.jpg kaydedildi.
M1002_img000359.jpg kaydedildi.
M1002_img000360.jpg kaydedildi.
M1002_img000361.jpg kaydedildi.
M1002_img000362.jpg kaydedildi.
M1002_img000363.jpg kaydedildi.
M1002_img000364.jpg kaydedildi.
M1002_img000365.jpg kaydedildi.
M1002_img000366.jpg kaydedildi.
M1002_img000367.jpg kaydedildi.
M1002_img000368.jpg kaydedildi.
M1002_img000369.jpg kaydedildi.
M1002_img000370.jpg kaydedildi.
M1002_img000371.jpg kayde

In [ ]:
import os
import zipfile

# Yeni klasör adı
new_ann_folder = '/content/drive/MyDrive/2209/medium_alt/medium_alt_annotations'

# Eğer klasör yoksa oluştur
if not os.path.exists(new_ann_folder):
    os.makedirs(new_ann_folder)

# Zip dosyasının yolu
zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

# Zip dosyasını aç ve JSON dosyalarını al
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for img_name in matching_images:
        # JSON dosyasının yolu
        ann_file_name = img_name.replace('.jpg', '.jpg.json')
        ann_file_path = f"UAVDT/train/ann/{ann_file_name}"

        # Eğer dosya zip dosyasında varsa
        if ann_file_path in zip_ref.namelist():
            # Dosyayı okuma
            with zip_ref.open(ann_file_path) as file:
                ann_data = file.read()

            # Yeni dosya yolunu oluştur
            new_ann_path = os.path.join(new_ann_folder, ann_file_name)

            # JSON dosyasını kaydet
            with open(new_ann_path, 'wb') as new_file:
                new_file.write(ann_data)

            print(f"{ann_file_name} kaydedildi.")

print(f"Tüm eşleşen JSON dosyaları '{new_ann_folder}' klasörüne kaydedildi.")

Görüntülenen çıkış son 5000 satıra kısaltıldı.
M1002_img000341.jpg.json kaydedildi.
M1002_img000342.jpg.json kaydedildi.
M1002_img000343.jpg.json kaydedildi.
M1002_img000344.jpg.json kaydedildi.
M1002_img000345.jpg.json kaydedildi.
M1002_img000346.jpg.json kaydedildi.
M1002_img000347.jpg.json kaydedildi.
M1002_img000348.jpg.json kaydedildi.
M1002_img000349.jpg.json kaydedildi.
M1002_img000350.jpg.json kaydedildi.
M1002_img000351.jpg.json kaydedildi.
M1002_img000352.jpg.json kaydedildi.
M1002_img000353.jpg.json kaydedildi.
M1002_img000354.jpg.json kaydedildi.
M1002_img000355.jpg.json kaydedildi.
M1002_img000356.jpg.json kaydedildi.
M1002_img000357.jpg.json kaydedildi.
M1002_img000358.jpg.json kaydedildi.
M1002_img000359.jpg.json kaydedildi.
M1002_img000360.jpg.json kaydedildi.
M1002_img000361.jpg.json kaydedildi.
M1002_img000362.jpg.json kaydedildi.
M1002_img000363.jpg.json kaydedildi.
M1002_img000364.jpg.json kaydedildi.
M1002_img000365.jpg.json kaydedildi.
M1002_img000366.jpg.json kay

## Low Alt

In [ ]:
import json

# "low alt" etiketi içeren dosyaları bul
low_alt_files = []

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for ann_file in train_ann_files:
        # JSON dosyasını oku
        with zip_ref.open(ann_file) as file:
            data = json.load(file)
            # "low alt" etiketini kontrol et
            if any(tag['name'] == 'low alt' for tag in data.get('tags', [])):
                low_alt_files.append(ann_file)

# "low alt" etiketine sahip dosyaları listele
print("Low alt etiketine sahip JSON dosyaları:", low_alt_files)
print(f"Toplam {len(low_alt_files)} JSON dosyası bulundu.")

KeyboardInterrupt: 

### dosyalardan .json uzantısı kaldırıldı.

In [ ]:
new_low_images= []

for ann_file in low_alt_files:
  image_name = ann_file.replace('UAVDT/train/ann/', '').replace('.jpg.json', '') + '.jpg'
  new_low_images.append(image_name)

print(f"Toplam {len(new_low_images)} dosya bulundu.")

### toplam görüntü içerisinde low_alt etiketine sahip görsel sayısı

In [ ]:
# "train/img" klasöründeki tüm dosyaları listele
img_files = []

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    img_files = [file for file in zip_ref.namelist() if 'train/img' in file and file.endswith('.jpg')]

# 'UAVDT/train/img/' kısmını her dosya isminden kaldır
img_file_names = [file.replace('UAVDT/train/img/', '') for file in img_files]

# Sonuçları yazdır
print(f"Toplam {len(img_file_names)} görsel ismi bulundu.")

matching_images = []

for img_name in new_low_images:
  if img_name in img_file_names:
    matching_images.append(img_name)

print(f"Toplam {len(matching_images)} dosya bulundu.")

### low alt etiketine sahip görseller için ayrı bir klasör oluştur

In [ ]:
import os
import zipfile
from io import BytesIO
from PIL import Image

# Yeni klasör adı
new_folder = '/content/drive/MyDrive/2209/low_alt'

# Eğer klasör yoksa oluştur
if not os.path.exists(new_folder):
    os.makedirs(new_folder)

# Zip dosyasının yolu
zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

# Zip dosyasını aç ve görselleri al
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for img_name in matching_images:
        # Görsel dosyasının yolu
        img_path = f"UAVDT/train/img/{img_name}"

        # Eğer dosya zip dosyasında varsa
        if img_path in zip_ref.namelist():
            # Dosyayı okuma
            with zip_ref.open(img_path) as file:
                img_data = file.read()

            # Görseli bellekte aç
            img = Image.open(BytesIO(img_data))

            # Yeni dosya yolunu oluştur
            new_img_path = os.path.join(new_folder, img_name)

            # Görseli kaydet
            img.save(new_img_path)

            print(f"{img_name} kaydedildi.")

print(f"Tüm eşleşen görseller '{new_folder}' klasörüne kaydedildi.")
print(f"Toplam {len(matching_images)} görsel kaydedildi.")

### json dosyaları için de ayrı klasör oluştur.

In [ ]:
import os
import zipfile

# Yeni klasör adı
new_ann_folder = '/content/drive/MyDrive/2209/low_alt_annotations'

# Eğer klasör yoksa oluştur
if not os.path.exists(new_ann_folder):
    os.makedirs(new_ann_folder)

# Zip dosyasının yolu
zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

# Zip dosyasını aç ve JSON dosyalarını al
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    for img_name in matching_images:
        # JSON dosyasının yolu
        ann_file_name = img_name.replace('.jpg', '.jpg.json')
        ann_file_path = f"UAVDT/train/ann/{ann_file_name}"

        # Eğer dosya zip dosyasında varsa
        if ann_file_path in zip_ref.namelist():
            # Dosyayı okuma
            with zip_ref.open(ann_file_path) as file:
                ann_data = file.read()

            # Yeni dosya yolunu oluştur
            new_ann_path = os.path.join(new_ann_folder, ann_file_name)

            # JSON dosyasını kaydet
            with open(new_ann_path, 'wb') as new_file:
                new_file.write(ann_data)

            print(f"{ann_file_name} kaydedildi.")

print(f"Tüm eşleşen JSON dosyaları '{new_ann_folder}' klasörüne kaydedildi.")

In [ ]:
import os

# Klasörlerin yolu
annotations_folder = '/content/drive/MyDrive/2209/low_alt_annotations'
images_folder = '/content/drive/MyDrive/2209/low_alt_images'

# Klasörlerdeki dosya isimlerini al
annotations_files = [f for f in os.listdir(annotations_folder) if os.path.isfile(os.path.join(annotations_folder, f))]
images_files = [f for f in os.listdir(images_folder) if os.path.isfile(os.path.join(images_folder, f))]

# 1. Klasörlerdeki dosya sayısını yazdır
print(f"low_alt_annotations klasöründe {len(annotations_files)} dosya bulundu.")
print(f"low_alt_images klasöründe {len(images_files)} dosya bulundu.")

## Hava Durumuna Göre Veri Setini Ayırmak

*   Görüntüleri "daylight", "night", "fog" olarak (hava durumuna göre) ayırmak

In [ ]:
import zipfile
import os

zip_file_path = '/content/drive/MyDrive/2209/UAVDT.zip'

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    file_list = zip_ref.namelist()
    train_ann_files = [file for file in file_list if 'train/ann' in file and file.endswith('.json')]

print(f"Toplam {len(train_ann_files)} JSON dosyası bulundu.")

Toplam 24143 JSON dosyası bulundu.


## Daylight etiketine sahip görsellerin bulunması.

In [ ]:
import json

daylight_files = []

with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
  for ann_file in train_ann_files:
    with zip_ref.open(ann_file) as file:
      data = json.load(file)
      if any(tag['name'] == 'daylight' for tag in data.get('tags', [])):
        daylight_files.append(ann_file)

print(f"Toplam {len(daylight_files)} JSON dosyası bulundu.")

Toplam 14679 JSON dosyası bulundu.


In [ ]:
# json uzantısını kaldır
new_daylight_images = []

for ann_file in daylight_files:
  image_name = ann_file.replace('UAVDT/train/ann/', '').replace('.jpg.json', '') + '.jpg'
  new_daylight_images.append(image_name)

print(f"Toplam {len(new_daylight_images)} dosya bulundu.")

Toplam 14679 dosya bulundu.


In [ ]:
img_files = []
with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
    img_files = [file for file in zip_ref.namelist() if 'train/img' in file and file.endswith('.jpg')]
img_file_names = [file.replace('UAVDT/train/img/', '') for file in img_files]

matching_names = []

for img_name in new_daylight_images:
  if img_name in img_file_names:
    matching_names.append(img_name)

print(f"Toplam {len(matching_names)} dosya bulundu.")

Toplam 14679 dosya bulundu.


In [15]:
import os
import shutil

image_train_dir = "/content/drive/MyDrive/2209/night/night_object_detection/images/train"
image_val_dir = "/content/drive/MyDrive/2209/night/night_object_detection/images/val"

label_source_dir = "/content/drive/MyDrive/2209/night/yolo_labels"
label_train_dir = "/content/drive/MyDrive/2209/night/night_object_detection/labels/train"
label_val_dir = "/content/drive/MyDrive/2209/night/night_object_detection/labels/val"

os.makedirs(label_train_dir, exist_ok=True)

train_images = os.listdir(image_train_dir)

copied_files = []

for img_file in train_images:
    img_name, _ = os.path.splitext(img_file)
    label_file = img_name + ".jpg.txt"

    src_path = os.path.join(label_source_dir, label_file)
    dest_path = os.path.join(label_train_dir, img_name + ".txt")

    if os.path.exists(src_path):
        shutil.copy(src_path, dest_path)
        copied_files.append(img_name + ".txt")

if copied_files:
    print("✅ Başarıyla kopyalanan dosyalar:")
else:
    print("⚠️ Hiçbir dosya kopyalanmadı. Dosya adlarını kontrol et!")

✅ Başarıyla kopyalanan dosyalar:


In [16]:
import os
import shutil

image_train_dir = "/content/drive/MyDrive/2209/night/night_object_detection/images/train"
image_val_dir = "/content/drive/MyDrive/2209/night/night_object_detection/images/val"

label_source_dir = "/content/drive/MyDrive/2209/night/yolo_labels"
label_train_dir = "/content/drive/MyDrive/2209/night/night_object_detection/labels/train"
label_val_dir = "/content/drive/MyDrive/2209/night/night_object_detection/labels/val"

os.makedirs(label_val_dir, exist_ok=True)

train_images = os.listdir(image_val_dir)

copied_files = []

for img_file in train_images:
    img_name, _ = os.path.splitext(img_file)
    label_file = img_name + ".jpg.txt"

    src_path = os.path.join(label_source_dir, label_file)
    dest_path = os.path.join(label_val_dir, img_name + ".txt")

    if os.path.exists(src_path):
        shutil.copy(src_path, dest_path)
        copied_files.append(img_name + ".txt")

if copied_files:
    print("✅ Başarıyla kopyalanan dosyalar:")
else:
    print("⚠️ Hiçbir dosya kopyalanmadı. Dosya adlarını kontrol et!")

✅ Başarıyla kopyalanan dosyalar:


In [17]:
yaml_content = """train: /content/drive/MyDrive/2209/night/night_object_detection/images/train
val: /content/drive/MyDrive/2209/night/night_object_detection/images/val

nc: 4
names: ["car", "vehicle", "truck", "bus"]
"""

yaml_path = "/content/drive/MyDrive/2209/night/night_object_detection/data.yaml"

with open(yaml_path, "w") as file:
    file.write(yaml_content)

print(f"✅ data.yaml başarıyla kaydedildi: {yaml_path}")

✅ data.yaml başarıyla kaydedildi: /content/drive/MyDrive/2209/night/night_object_detection/data.yaml


In [19]:
!cat /content/drive/MyDrive/2209/night/night_object_detection/data.yaml

train: /content/drive/MyDrive/2209/night/night_object_detection/images/train
val: /content/drive/MyDrive/2209/night/night_object_detection/images/val

nc: 4
names: ["car", "vehicle", "truck", "bus"]


## Daylight - Train olarak ayırmak

In [20]:
import os
import shutil
import random

image_dir = "/content/drive/MyDrive/2209/daylight/daylight_images"
label_dir = "/content/drive/MyDrive/2209/daylight/yolo_labels"

train_img_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/images/train"
val_img_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/images/val"
train_label_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/labels/train"
val_label_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/labels/val"

os.makedirs(train_img_dir, exist_ok=True)
os.makedirs(val_img_dir, exist_ok=True)
os.makedirs(train_label_dir, exist_ok=True)
os.makedirs(val_label_dir, exist_ok=True)

image_files = [f for f in os.listdir(image_dir) if f.endswith(".jpg") or f.endswith(".png")]

random.shuffle(image_files)
split_index = int(len(image_files) * 0.8)
train_files = image_files[:split_index]
val_files = image_files[split_index:]

for file in train_files:
    shutil.move(os.path.join(image_dir, file), os.path.join(train_img_dir, file))
    label_file = file.replace(".jpg", ".txt").replace(".png", ".txt")
    if os.path.exists(os.path.join(label_dir, label_file)):
        shutil.move(os.path.join(label_dir, label_file), os.path.join(train_label_dir, label_file))

for file in val_files:
    shutil.move(os.path.join(image_dir, file), os.path.join(val_img_dir, file))
    label_file = file.replace(".jpg", ".txt").replace(".png", ".txt")
    if os.path.exists(os.path.join(label_dir, label_file)):
        shutil.move(os.path.join(label_dir, label_file), os.path.join(val_label_dir, label_file))

print("✅ Veri seti başarıyla train ve val olarak ayrıldı!")

✅ Veri seti başarıyla train ve val olarak ayrıldı!


In [21]:
import os
import shutil

image_train_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/images/train"
image_val_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/images/val"

label_source_dir = "/content/drive/MyDrive/2209/daylight/yolo_labels"
label_train_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/labels/train"
label_val_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/labels/val"

os.makedirs(label_train_dir, exist_ok=True)

train_images = os.listdir(image_train_dir)

copied_files = []

for img_file in train_images:
    img_name, _ = os.path.splitext(img_file)
    label_file = img_name + ".jpg.txt"

    src_path = os.path.join(label_source_dir, label_file)
    dest_path = os.path.join(label_train_dir, img_name + ".txt")

    if os.path.exists(src_path):
        shutil.copy(src_path, dest_path)
        copied_files.append(img_name + ".txt")

if copied_files:
    print("✅ Başarıyla kopyalanan dosyalar:")
else:
    print("⚠️ Hiçbir dosya kopyalanmadı. Dosya adlarını kontrol et!")


image_train_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/images/train"
image_val_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/images/val"

label_source_dir = "/content/drive/MyDrive/2209/daylight/yolo_labels"
label_train_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/labels/train"
label_val_dir = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/labels/val"

os.makedirs(label_val_dir, exist_ok=True)

train_images = os.listdir(image_val_dir)

copied_files = []

for img_file in train_images:
    img_name, _ = os.path.splitext(img_file)
    label_file = img_name + ".jpg.txt"

    src_path = os.path.join(label_source_dir, label_file)
    dest_path = os.path.join(label_val_dir, img_name + ".txt")

    if os.path.exists(src_path):
        shutil.copy(src_path, dest_path)
        copied_files.append(img_name + ".txt")

if copied_files:
    print("✅ Başarıyla kopyalanan dosyalar:")
else:
    print("⚠️ Hiçbir dosya kopyalanmadı. Dosya adlarını kontrol et!")

✅ Başarıyla kopyalanan dosyalar:
✅ Başarıyla kopyalanan dosyalar:


In [22]:
yaml_content = """train: /content/drive/MyDrive/2209/daylight/daylight_object_detection/images/train
val: /content/drive/MyDrive/2209/daylight/daylight_object_detection/images/val

nc: 4
names: ["car", "vehicle", "truck", "bus"]
"""

yaml_path = "/content/drive/MyDrive/2209/daylight/daylight_object_detection/data.yaml"

with open(yaml_path, "w") as file:
    file.write(yaml_content)

print(f"✅ data.yaml başarıyla kaydedildi: {yaml_path}")

✅ data.yaml başarıyla kaydedildi: /content/drive/MyDrive/2209/daylight/daylight_object_detection/data.yaml


In [24]:
import os
import shutil
import random

image_dir = "/content/drive/MyDrive/2209/fog/images"
label_dir = "/content/drive/MyDrive/2209/fog/yolo_labels"

train_img_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/images/train"
val_img_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/images/val"
train_label_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/labels/train"
val_label_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/labels/val"

os.makedirs(train_img_dir, exist_ok=True)
os.makedirs(val_img_dir, exist_ok=True)
os.makedirs(train_label_dir, exist_ok=True)
os.makedirs(val_label_dir, exist_ok=True)

image_files = [f for f in os.listdir(image_dir) if f.endswith(".jpg") or f.endswith(".png")]

random.shuffle(image_files)
split_index = int(len(image_files) * 0.8)
train_files = image_files[:split_index]
val_files = image_files[split_index:]

for file in train_files:
    shutil.move(os.path.join(image_dir, file), os.path.join(train_img_dir, file))
    label_file = file.replace(".jpg", ".txt").replace(".png", ".txt")
    if os.path.exists(os.path.join(label_dir, label_file)):
        shutil.move(os.path.join(label_dir, label_file), os.path.join(train_label_dir, label_file))

for file in val_files:
    shutil.move(os.path.join(image_dir, file), os.path.join(val_img_dir, file))
    label_file = file.replace(".jpg", ".txt").replace(".png", ".txt")
    if os.path.exists(os.path.join(label_dir, label_file)):
        shutil.move(os.path.join(label_dir, label_file), os.path.join(val_label_dir, label_file))

print("✅ Veri seti başarıyla train ve val olarak ayrıldı!")

✅ Veri seti başarıyla train ve val olarak ayrıldı!


In [25]:
import os
import shutil

image_train_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/images/train"
image_val_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/images/val"

label_source_dir = "/content/drive/MyDrive/2209/fog/yolo_labels"
label_train_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/labels/train"
label_val_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/labels/val"

os.makedirs(label_train_dir, exist_ok=True)

train_images = os.listdir(image_train_dir)

copied_files = []

for img_file in train_images:
    img_name, _ = os.path.splitext(img_file)
    label_file = img_name + ".jpg.txt"

    src_path = os.path.join(label_source_dir, label_file)
    dest_path = os.path.join(label_train_dir, img_name + ".txt")

    if os.path.exists(src_path):
        shutil.copy(src_path, dest_path)
        copied_files.append(img_name + ".txt")

if copied_files:
    print("✅ Başarıyla kopyalanan dosyalar:")
else:
    print("⚠️ Hiçbir dosya kopyalanmadı. Dosya adlarını kontrol et!")

image_train_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/images/train"
image_val_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/images/val"

label_source_dir = "/content/drive/MyDrive/2209/fog/yolo_labels"
label_train_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/labels/train"
label_val_dir = "/content/drive/MyDrive/2209/fog/fog_object_detection/labels/val"

os.makedirs(label_val_dir, exist_ok=True)

train_images = os.listdir(image_val_dir)

copied_files = []

for img_file in train_images:
    img_name, _ = os.path.splitext(img_file)
    label_file = img_name + ".jpg.txt"

    src_path = os.path.join(label_source_dir, label_file)
    dest_path = os.path.join(label_val_dir, img_name + ".txt")

    if os.path.exists(src_path):
        shutil.copy(src_path, dest_path)
        copied_files.append(img_name + ".txt")

if copied_files:
    print("✅ Başarıyla kopyalanan dosyalar:")
else:
    print("⚠️ Hiçbir dosya kopyalanmadı. Dosya adlarını kontrol et!")


✅ Başarıyla kopyalanan dosyalar:
✅ Başarıyla kopyalanan dosyalar:


In [26]:
yaml_content = """train: /content/drive/MyDrive/2209/fog/fog_object_detection/images/train
val: /content/drive/MyDrive/2209/fog/fog_object_detection/images/val

nc: 4
names: ["car", "vehicle", "truck", "bus"]
"""

yaml_path = "/content/drive/MyDrive/2209/fog/fog_object_detection/data.yaml"

with open(yaml_path, "w") as file:
    file.write(yaml_content)

print(f"✅ data.yaml başarıyla kaydedildi: {yaml_path}")

✅ data.yaml başarıyla kaydedildi: /content/drive/MyDrive/2209/fog/fog_object_detection/data.yaml


In [27]:
!cat /content/drive/MyDrive/2209/fog/fog_object_detection/data.yaml

train: /content/drive/MyDrive/2209/fog/fog_object_detection/images/train
val: /content/drive/MyDrive/2209/fog/fog_object_detection/images/val

nc: 4
names: ["car", "vehicle", "truck", "bus"]
